tests/test_eclinterregflows.cpp

```diff
@@ -0,0 +1,3413 @@
+            bool isInterior(const int cellID)
+            {
+                return cellID == 2;
+            }
+        } // namespace P2
+
+        namespace P3 {
+            bool isInterior(const int cellID)
+            {
+                return cellID == 3;
+            }
+        } // namespace P3
+
+        namespace P4 {
+            bool isInterior(const int cellID)
+            {
+                return cellID == 0;
+            }
+        } // namespace P4
+    } // namespace FourProc
+}
+
+// =====================================================================
+
+BOOST_AUTO_TEST_SUITE(Single_FIP_Region)
+
+BOOST_AUTO_TEST_CASE(AllInternal)
+{
+    auto flows = Opm::EclInterRegFlowMapSingleFIP{ all_same_region() };
+    BOOST_CHECK_EQUAL(flows.getLocalMaxRegionID(), std::size_t{1});
+
+    BOOST_CHECK_MESSAGE(flows.assignGlobalMaxRegionID(5),
+                        "Assigning large global maximum "
+                        "Region ID must succeed");
+
+    flows.addConnection({ 0, true }, { 1, true }, conn_01());
+    flows.addConnection({ 0, true }, { 2, true }, conn_02());
+    flows.addConnection({ 1, true }, { 3, true }, conn_13());
+    flows.addConnection({ 2, true }, { 3, true }, conn_23());
+
+    flows.compress();
+
+    const auto& iregFlows = flows.getInterRegFlows();
+    BOOST_CHECK_EQUAL(iregFlows.numRegions(), 5);
+
+    for (auto r1 = 0*iregFlows.numRegions(); r1 < iregFlows.numRegions(); ++r1) {
+        for (auto r2 = r1 + 1; r2 < iregFlows.numRegions(); ++r2) {
+            auto flow = iregFlows.getInterRegFlows(r1, r2);
+            BOOST_CHECK_MESSAGE(! flow.has_value(),
+                                "There must not be inter-regional flow "
+                                "between regions " << r1 << " and " << r2);
+        }
+    }
+}
+
+BOOST_AUTO_TEST_CASE(LeftInternal_RightOther)
+{
+    auto flows = Opm::EclInterRegFlowMapSingleFIP{ all_same_region() };
+    BOOST_CHECK_EQUAL(flows.getLocalMaxRegionID(), std::size_t{1});
+
+    BOOST_CHECK_MESSAGE(flows.assignGlobalMaxRegionID(5),
+                        "Assigning large global maximum "
+                        "Region ID must succeed");
+
+    flows.addConnection({ 0, true } , { 1, false }, conn_01());
+    flows.addConnection({ 0, true } , { 2, true } , conn_02());
+    flows.addConnection({ 1, false }, { 3, false }, conn_13());
+    flows.addConnection({ 2, true } , { 3, false }, conn_23());
+
+    flows.compress();
+
+    const auto& iregFlows = flows.getInterRegFlows();
+    BOOST_CHECK_EQUAL(iregFlows.numRegions(), 5);
+
+    for (auto r1 = 0*iregFlows.numRegions(); r1 < iregFlows.numRegions(); ++r1) {
+        for (auto r2 = r1 + 1; r2 < iregFlows.numRegions(); ++r2) {
+            auto flow = iregFlows.getInterRegFlows(r1, r2);
+            BOOST_CHECK_MESSAGE(! flow.has_value(),
+                                "There must not be inter-regional flow "
+                                "between regions " << r1 << " and " << r2);
+        }
```

Comment:
Shouldn't we also check that flow rate is correct?

Reply:
> Shouldn't we also check that the flow rate is correct?

Not here.  This particular check verifies that there is no flow at all between the two regions.